File: library/src/main/java/com/alamkanak/weekview/WeekViewLoader.java
package com.alamkanak.weekview;

import java.util.Calendar;
import java.util.List;

public interface WeekViewLoader {
    /**
     * Convert a date into a double that will be used to reference when you're loading data.
     *
     * All periods that have the same integer part, define one period. Dates that are later in time should have a greater return value.
     *
     * @param instance the date
     * @return The period index in which the date falls (floating point number).
     */
    public double toWeekViewPeriodIndex(Calendar instance);

    /**
     * Load the events within the period
     * @param periodIndex the period to load
     * @return A list with the events of this period
     */
    public List<WeekViewEvent> onLoad(int periodIndex);
}

File: library/src/main/java/com/alamkanak/weekview/WeekViewLoader.kt
package com.alamkanak.weekview

import java.util.*

interface WeekViewLoader<T> {

    /**
     * Converts a date into a [Period] that will be used for reference when you're loading data.
     *
     * @param instance The date
     * @return The [Period] in which the date falls
     */
    fun toPeriod(instance: Calendar): Period

    /**
     * Loads the events within the specified period
     *
     * @param period The [Period] to load
     * @return A list with the [WeekViewEvent]s of this [Period]
     */
    fun onLoad(period: Period): List<WeekViewEvent<T>>

}
